### PR TITLE
Refactor MSALTokenCache

### PR DESCRIPTION
**Pull request by unpluggedk**
- Remove MSALRequestParameter from MSALTokenCache methods
- Split saveAccessAndRefreshToken into 2 methods

Addresses #76 and #77

### CODE REVIEW

File: MSAL/src/cache/MSALTokenCache.m [hunk 1]
```diff
@@ -158,11 +178,11 @@ - (MSALAccessTokenCacheItem *)findAccessToken:(MSALRequestParameters *)requestPa
     
     for (MSALAccessTokenCacheItem *tokenItem in allAccessTokens)
     {
-        if (requestParam.unvalidatedAuthority && [key matches:[tokenItem tokenCacheKey:nil]])
+        if (authority && [key matches:[tokenItem tokenCacheKey:nil]])
```
Comment (RPangrle) [inline MSAL/src/cache/MSALTokenCache.m]: This line worries me. There's more to a token cache key then authority...

**Comment by jasoncoolmax** [comment]
I have changed the class names:
`MSALTokenCacheAccessor`     -> `MSALTokenCache`
`MSALTokenCacheDataSource` -> `MSALTokenCacheAccessor`,
which caused confusion at that time.

@RPangrle Do we still need to address those issues (#76 and #77)?  As we are already aligned with .Net and Android for TokenCache APIs:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/74b53690c328eb6611146a0a5dfb798b340a10ab/src/Microsoft.Identity.Client/TokenCache.cs#L104 

If we want to make the changes, I am also OK :)

**Comment by RPangrle** [comment]
Those issues I filed were also to try to cleanup the caching logic. Our two goals are to 1) write clean code and 2) match the object model is the C# code as much as is reasonable. If you find places where #2 seems in conflict with #1 then it's a good topic for conversation and filing issues. :)

File: MSAL/src/cache/MSALTokenCache.m (hunk 1 above)
Comment (unpluggedk) [inline MSAL/src/cache/MSALTokenCache.m]: I've moved things around, no more need for MSALAccessTokenCacheKey and [key matches:]

Things checked:
- scopes: checked as a first condition to match
- clientId: filtered when allAccessTokensForUser: is called
- authority: checked after scopes.
- user: filtered when allAccessTokensForUser: is called

**Pull request merged**